// framer-motion-carousel/dist/cjs/arrow.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var baseArrowStyle = {
  position: "absolute",
  width: "50px",
  height: "50px",
  backgroundColor: "rgba(0,0,0,0.5)",
  top: "50%",
  transform: "translateY(-50%)",
  borderRadius: "50%",
  color: "#fff",
  fontSize: "20px",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  cursor: "pointer"
};

var Arrow = function Arrow(_ref) {
  var _ref$left = _ref.left,
      left = _ref$left === void 0 ? false : _ref$left,
      children = _ref.children,
      onClick = _ref.onClick;
  return /*#__PURE__*/_react["default"].createElement("div", {
    onClick: onClick,
    style: _extends({}, baseArrowStyle, {
      left: left ? "20px" : "initial",
      right: !left ? "10px" : "initial"
    })
  }, children);
};

var _default = Arrow;
exports["default"] = _default;
