// framer-motion-carousel/dist/cjs/carousel.js
"use strict";

exports.__esModule = true;
exports.Carousel = void 0;

var React = _interopRequireWildcard(require("react"));

var _framerMotion = require("framer-motion");

var _arrow = _interopRequireDefault(require("./arrow"));

var _slider = _interopRequireDefault(require("./slider"));

var _dots = _interopRequireDefault(require("./dots"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var containerStyle = {
  position: "relative",
  width: "100%",
  height: "100%",
  overflowX: "hidden",
  display: "flex"
};
var transition = {
  type: "spring",
  bounce: 0
};
var Contaier = /*#__PURE__*/React.forwardRef(function (props, ref) {
  return /*#__PURE__*/React.createElement("div", {
    ref: ref,
    style: containerStyle
  }, props.children);
});

var Carousel = function Carousel(_ref) {
  var children = _ref.children,
      renderArrowLeft = _ref.renderArrowLeft,
      renderArrowRight = _ref.renderArrowRight,
      renderDots = _ref.renderDots,
      _ref$autoPlay = _ref.autoPlay,
      autoPlay = _ref$autoPlay === void 0 ? true : _ref$autoPlay,
      _ref$interval = _ref.interval,
      interval = _ref$interval === void 0 ? 2000 : _ref$interval,
      _ref$loop = _ref.loop,
      loop = _ref$loop === void 0 ? true : _ref$loop;
  var x = (0, _framerMotion.useMotionValue)(0);
  var containerRef = React.useRef(null);

  var _React$useState = React.useState(0),
      index = _React$useState[0],
      setIndex = _React$useState[1];

  var calculateNewX = function calculateNewX() {
    var _containerRef$current;

    return -index * (((_containerRef$current = containerRef.current) == null ? void 0 : _containerRef$current.clientWidth) || 0);
  };

  var handleEndDrag = function handleEndDrag(e, dragProps) {
    var _containerRef$current2;

    var clientWidth = ((_containerRef$current2 = containerRef.current) == null ? void 0 : _containerRef$current2.clientWidth) || 0;
    var offset = dragProps.offset;

    if (offset.x > clientWidth / 4) {
      handlePrev();
    } else if (offset.x < -clientWidth / 4) {
      handleNext();
    } else {
      (0, _framerMotion.animate)(x, calculateNewX(), transition);
    }
  };

  var childrens = React.Children.toArray(children);

  var handleNext = function handleNext() {
    var idx = loop ? 0 : index;
    setIndex(index + 1 === childrens.length ? idx : index + 1);
  };

  var handlePrev = function handlePrev() {
    var idx = loop ? childrens.length - 1 : 0;
    setIndex(index - 1 < 0 ? idx : index - 1);
  };

  React.useEffect(function () {
    var controls = (0, _framerMotion.animate)(x, calculateNewX(), transition);
    return controls.stop;
  }, [index]);
  React.useEffect(function () {
    if (!autoPlay) {
      return;
    }

    var timer = setInterval(function () {
      return handleNext();
    }, interval);
    return function () {
      return clearInterval(timer);
    };
  }, [handleNext, interval]);
  return /*#__PURE__*/React.createElement(Contaier, {
    ref: containerRef
  }, childrens.map(function (child, i) {
    return /*#__PURE__*/React.createElement(_slider["default"], {
      onDragEnd: handleEndDrag,
      x: x,
      i: i,
      key: i
    }, child);
  }), renderArrowLeft ? renderArrowLeft({
    handlePrev: handlePrev,
    activeIndex: index
  }) : /*#__PURE__*/React.createElement(_arrow["default"], {
    left: true,
    onClick: handlePrev
  }, "\u2190"), renderArrowRight ? renderArrowRight({
    handleNext: handleNext,
    activeIndex: index
  }) : /*#__PURE__*/React.createElement(_arrow["default"], {
    onClick: handleNext
  }, "\u2192"), renderDots ? renderDots({
    setActiveIndex: setIndex,
    activeIndex: index
  }) : /*#__PURE__*/React.createElement(_dots["default"], {
    length: childrens.length,
    setActiveIndex: setIndex,
    activeIndex: index
  }));
};

exports.Carousel = Carousel;
