// framer-motion-carousel/dist/cjs/dots.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var dotWrapStyle = {
  position: "absolute",
  bottom: "10px",
  left: "50%",
  transform: "translateX(-50%)"
};
var dotItemStyle = {
  width: "20px",
  height: "20px",
  borderRadius: "50%",
  margin: "0 10px",
  display: "inline-block",
  cursor: "pointer"
};

var Dots = function Dots(_ref) {
  var length = _ref.length,
      activeIndex = _ref.activeIndex,
      setActiveIndex = _ref.setActiveIndex;
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: dotWrapStyle
  }, new Array(length).fill("").map(function (_, i) {
    return /*#__PURE__*/_react["default"].createElement("span", {
      onClick: function onClick() {
        return setActiveIndex(i);
      },
      key: i,
      style: _extends({}, dotItemStyle, {
        background: i === activeIndex ? "#000" : "#999",
        transform: "scale(" + (i === activeIndex ? 1.3 : 1) + ")"
      })
    });
  }));
};

var _default = Dots;
exports["default"] = _default;
