// framer-motion-carousel/dist/cjs/slider.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _framerMotion = require("framer-motion");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var pageStyle = {
  width: "100%",
  height: "100%",
  display: "inline-block",
  flex: "none"
};

var Slider = function Slider(_ref) {
  var x = _ref.x,
      i = _ref.i,
      onDragEnd = _ref.onDragEnd,
      children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement(_framerMotion.motion.div, {
    style: _extends({}, pageStyle, {
      x: x,
      left: i * 100 + "%",
      right: i * 100 + "%"
    }),
    drag: "x",
    dragElastic: 0.3,
    onDragEnd: onDragEnd
  }, children);
};

var _default = Slider;
exports["default"] = _default;
